### PR TITLE
Added method to log in WAL custom WALRecords

### PR DESCRIPTION
**Pull request by wolf4ood**

### CODE REVIEW

File: core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java [hunk 1]
```diff
@@ -52,6 +52,7 @@
   private Map<String, Long>      newFileNamesId       = new HashMap<String, Long>();
   private Set<Long>              deletedFiles         = new HashSet<Long>();
   private Map<String, Long>      deletedFileNameIdMap = new HashMap<String, Long>();
+  private List<OWALRecord>       externalLogs         = new ArrayList<OWALRecord>();
```
Comment (laa) [inline core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java]: That is for develop branch as I can see, could you use diamond operator instead  ?))

File: core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java [hunk 2]
```diff
@@ -169,6 +168,14 @@ public void pinPage(OCacheEntry cacheEntry) throws IOException {
     pageChangesContainer.pinPage = true;
   }
 
+  /**
+   * Add External WalRecord to the atomic operation
+   * @param record
```
Comment (laa) [inline core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java]: The empty parameter, javadoc, could you remove it.

File: core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java [hunk 3]
```diff
@@ -380,7 +387,9 @@ else if (fileChanges.truncate)
           }
         }
       }
-
+      for (OWALRecord externalLog : externalLogs) {
+        writeAheadLog.log(externalLog);
```
Comment (laa) [inline core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java]: Write ahead log can be null, this call can produce NPE, you should move it one line above.

File: core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java (hunk 1 above)
Comment (maggiolo00) [reply core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java]: :)

File: core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java (hunk 2 above)
Comment (maggiolo00) [reply core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/atomicoperations/OAtomicOperation.java]: what do you mean empty parameter?

File: core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/wal/OWALRecordsFactory.java [hunk 4]
```diff
@@ -31,7 +34,16 @@
...
 
-  public static final OWALRecordsFactory INSTANCE    = new OWALRecordsFactory();
+  public static final OWALRecordsFactory INSTANCE;
+  static {
+
+    INSTANCE  = new OWALRecordsFactory();
+    final Iterator<OWALRecordsFactoryExternal> iterator = lookupProviderWithOrientClassLoader(OWALRecordsFactoryExternal.class, OWALRecordsFactory.class.getClassLoader());
+
+    while (iterator.hasNext()){
+      iterator.next().init(INSTANCE);
```
[the remark this answers is not in the document]
Comment (maggiolo00) [reply core/src/main/java/com/orientechnologies/orient/core/storage/impl/local/paginated/wal/OWALRecordsFactory.java]: Do i thrown an exception if present?

**Pull request closed without merging**